waku/waku_api/rest/filter/handlers.nim

```diff
@@ -159,14 +161,23 @@ proc filterPostPutSubscriptionRequestHandler(node: WakuNode,
 
   let req: FilterSubscribeRequest = decodedBody.value()
 
-  let peerOpt = node.peerManager.selectPeer(WakuFilterSubscribeCodec)
+  let peer = node.peerManager.selectPeer(WakuFilterSubscribeCodec).valueOr:
+    let handler = discHandler.valueOr:
+      return makeRestResponse(
+        req.requestId,
+        FilterSubscribeError.serviceUnavailable("No suitable peers")
```

Comment:
Yes, there's no service peers and no discovery handler and so the final result is "no suitable peer"

Reply:
Okay makes sense :)
I actually asked that because we bring the very same response a few lines below. I think is interesting to differentiate the response somehow to ease debugging future issues.

Reply:
Will do!